Build android with `ergo-protocol-client` dependency

Android build for wallet fails if we include `ergo-protocol-client` as dependency of `ergvein-wallet` package:

```
Linking dist/build/ergo-protocol-client/ergo-protocol-client ...
clang: warning: argument unused during compilation: '-no-pie' [-Wunused-command-line-argument]
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: error: /nix/store/jq6xr054gbb172dcp4cxn96c7bbf6ab6-ndk-bundle-21.0.6113669/libexec/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/armv7-a/libgcc.a(_arm_addsubdf3.o): multiple definition of '__aeabi_dsub'
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: /nix/store/f2chcx5kbgya204lsqsz9b05vqli4gz4-ergo-protocol-0.1.0.0-armv7a-unknown-linux-androideabi/lib/ghc-8.6.5/arm-android-ghc-8.6.5/ergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY/libHSergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY.a(compiler_builtins-ebd64c56e1c3cc00.compiler_builtins.4db5i49j-cgu.172.rcgu.o): previous definition here
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: error: /nix/store/jq6xr054gbb172dcp4cxn96c7bbf6ab6-ndk-bundle-21.0.6113669/libexec/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/armv7-a/libgcc.a(_arm_addsubdf3.o): multiple definition of '__aeabi_dadd'
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: /nix/store/f2chcx5kbgya204lsqsz9b05vqli4gz4-ergo-protocol-0.1.0.0-armv7a-unknown-linux-androideabi/lib/ghc-8.6.5/arm-android-ghc-8.6.5/ergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY/libHSergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY.a(compiler_builtins-ebd64c56e1c3cc00.compiler_builtins.4db5i49j-cgu.88.rcgu.o): previous definition here
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: error: /nix/store/jq6xr054gbb172dcp4cxn96c7bbf6ab6-ndk-bundle-21.0.6113669/libexec/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/armv7-a/libgcc.a(_arm_addsubdf3.o): multiple definition of '__aeabi_f2d'
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: /nix/store/f2chcx5kbgya204lsqsz9b05vqli4gz4-ergo-protocol-0.1.0.0-armv7a-unknown-linux-androideabi/lib/ghc-8.6.5/arm-android-ghc-8.6.5/ergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY/libHSergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY.a(compiler_builtins-ebd64c56e1c3cc00.compiler_builtins.4db5i49j-cgu.143.rcgu.o): previous definition here
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: error: /nix/store/jq6xr054gbb172dcp4cxn96c7bbf6ab6-ndk-bundle-21.0.6113669/libexec/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/armv7-a/libgcc.a(_arm_addsubdf3.o): multiple definition of '__aeabi_ul2d'
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: /nix/store/f2chcx5kbgya204lsqsz9b05vqli4gz4-ergo-protocol-0.1.0.0-armv7a-unknown-linux-androideabi/lib/ghc-8.6.5/arm-android-ghc-8.6.5/ergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY/libHSergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY.a(compiler_builtins-ebd64c56e1c3cc00.compiler_builtins.4db5i49j-cgu.146.rcgu.o): previous definition here
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: error: /nix/store/jq6xr054gbb172dcp4cxn96c7bbf6ab6-ndk-bundle-21.0.6113669/libexec/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/armv7-a/libgcc.a(_fixunsdfdi.o): multiple definition of '__aeabi_d2ulz'
/nix/store/bafbi20qwp73v6x0m5n1hn381h4z56ma-ndk-gcc-binutils/bin/armv7a-unknown-linux-androideabi-ld.gold: /nix/store/f2chcx5kbgya204lsqsz9b05vqli4gz4-ergo-protocol-0.1.0.0-armv7a-unknown-linux-androideabi/lib/ghc-8.6.5/arm-android-ghc-8.6.5/ergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY/libHSergo-protocol-0.1.0.0-3MRY5Idb6N0GDgRPROZjnY.a(compiler_builtins-ebd64c56e1c3cc00.compiler_builtins.4db5i49j-cgu.113.rcgu.o): previous definition here
[ 2 of 68] Compiling Ergvein.Core.Node.Types ( src/Ergvein/Core/Node/Types.hs, dist/build/Ergvein/Core/Node/Types.o )
[ 3 of 68] Compiling Ergvein.Core.Node.Monad ( src/Ergvein/Core/Node/Monad.hs, dist/build/Ergvein/Core/Node/Monad.o )
clang: error: linker command failed with exit code 1 (use -v to see invocation)
`armv7a-unknown-linux-androideabi-cc' failed in phase `Linker'. (Exit code: 1)
builder for '/nix/store/z5whplzqh01ni6rbvjzdhhsa8vsbjzsr-ergo-protocol-client-0.1.0.0-armv7a-unknown-linux-androideabi.drv' failed with exit code 1

```

Possible related: https://github.com/rust-lang/compiler-builtins/issues/353

--- a/ergo-protocol-client/ergo-protocol-client.cabal
+++ b/ergo-protocol-client/ergo-protocol-client.cabal
@@ -12,6 +12,12 @@ author:              Anton Gushcha, Aminion, Vladimir Krutkin, Levon Oganyan, Se
 maintainer:          Anton Gushcha <ncrashed@protonmail.com>
 extra-source-files:  CHANGELOG.md
 
+flag client-tool
+  description: Build executable for connecting to Ergo node
+  -- See #968, don't build on ARM to prevent linker errors
+  default: False
+  manual: True
+
 library
   hs-source-dirs:      src
   exposed-modules:
@@ -40,6 +46,10 @@ library
     TemplateHaskell
 
 executable ergo-protocol-client
+  if flag(client-tool)
+    buildable: True
+  else
+    buildable: False
   hs-source-dirs:      app
   main-is:             Main.hs
   ghc-options:         -threaded -O3

--- a/wallet/ergvein-wallet.cabal
+++ b/wallet/ergvein-wallet.cabal
@@ -115,6 +115,7 @@ library
         , directory
         , dns                     >= 4.0        && < 5.0
         , ergo-protocol
+        , ergo-protocol-client
         , ergvein-common          >= 1.0        && < 1.1
         , ergvein-core
         , ergvein-crypto          >= 0.1        && < 0.2

